jbpm-test/src/main/java/org/jbpm/test/AbstractBaseTest.java

```diff
@@ -16,19 +16,56 @@
 
 package org.jbpm.test;
 
+import java.util.Properties;
+
 import org.jbpm.process.instance.impl.util.LoggingPrintStream;
+import org.jbpm.test.persistence.util.PersistenceUtil;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
+import org.kie.test.util.db.PoolingDataSourceWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public abstract class AbstractBaseTest {
-    
+
+    private static final Logger logger = LoggerFactory.getLogger(AbstractBaseTest.class);
+
+    protected static String datasourceName = "jdbc/jbpm-ds";
+
     @BeforeClass
-    public static void configure() { 
+    public static void configure() {
         LoggingPrintStream.interceptSysOutSysErr();
     }
     
     @AfterClass
-    public static void reset() { 
+    public static void reset() {
         LoggingPrintStream.resetInterceptSysOutSysErr();
     }
+
+    protected String getJndiDatasourceName(){
+        return datasourceName;
+    }
+
+    protected Properties getDataSourceProperties(){
+        return PersistenceUtil.getDatasourceProperties();
+    }
+
+    protected static PoolingDataSourceWrapper setupPoolingDataSource(String datasourceName) {
+        return setupPoolingDataSource(null, datasourceName);
+    }
+
+    protected static PoolingDataSourceWrapper setupPoolingDataSource(Properties properties, String datasourceName) {
```

Comment:
Can't we directly use the PersistenceUtil class? We should always use the properties from datasource.properties file so I would delete the option of providing null properties completely.

Reply:
Good point. Done.